test/ttmlir/Dialect/TTNN/ccl/all_gather.mlir

```diff
@@ -2,10 +2,9 @@
 #any_device = #tt.operand_constraint<dram|l1|scalar|tile|any_device|any_device_tile>
 module attributes {} {
   func.func @forward(%arg0: tensor<1x1x32x32xbf16>) -> tensor<1x1x32x128xbf16> {
-    // CHECK: %[[C:.*]] = "ttnn.empty"[[C:.*]]
     %0 = tensor.empty() : tensor<1x1x32x128xbf16>
-    // CHECK: %[[C:.*]] = "ttnn.all_gather"[[C:.*]]
     %1 = "ttir.all_gather"(%arg0, %0) <{dim = 3 : si32, operand_constraints = [#any_device, #any_device]}> : (tensor<1x1x32x32xbf16>, tensor<1x1x32x128xbf16>) -> tensor<1x1x32x128xbf16>
+    // CHECK: %[[C:.*]] = "ttnn.all_gather"[[C:.*]]
```

Comment:
minor issue: should this line be above "ttir.all_gather"? FileCheck passes both ways but I think it would be appropriate to put it above.

Btw, conflict warning: I'm ready to merge https://github.com/tenstorrent/tt-mlir/pull/1182.

Reply:
@gfengTT  This unit test lowers ttir to ttnn, so I believe ttnn should be checked as a output of this pipeline, --ttir-to-ttnn-backend-pipeline. 

Please go ahead to merge #1182 . I manually applied your patch in this PR, so I will try to merge yours once it is available in the main branch.